apps/voting/contracts/Voting.sol

```diff
@@ -422,8 +580,73 @@ contract Voting is IForwarder, AragonApp {
         emit ExecuteVote(_voteId);
     }
 
+    /**
+     * @dev internal function to add the `_voter` to the list of delegated voters for the `_delegate` and update the delegate for the `_voter`
+     * @param _delegate address of the delegate
+     * @param _voter address of the voter
+     */
+    function _addDelegatedAddressFor(address _delegate, address _voter) internal {
+        uint256 delegatedVotersCount = delegatedVoters[_delegate].addresses.length;
+        require(delegatedVotersCount <= UINT_96_MAX, ERROR_MAX_DELEGATED_VOTERS_REACHED);
+
+        delegatedVoters[_delegate].addresses.push(_voter);
+        delegates[_voter] = Delegate(_delegate, uint96(delegatedVotersCount));
+        emit SetDelegate(_voter, _delegate);
+    }
+
+    /**
+     * @dev internal function to remove  the `_voter` from the list of delegated voters for the `_delegate`
+     * @param _delegate address of the delegate
+     * @param _voter address of the voter
+     */
+    function _removeDelegatedAddressFor(address _delegate, address _voter) internal {
+        uint256 delegatedVotersCount = delegatedVoters[_delegate].addresses.length;
+        require(delegatedVotersCount > 0, ERROR_DELEGATE_NOT_SET);
+
+        uint96 voterIndex = delegates[_voter].voterIndex;
+        assert(delegatedVoters[_delegate].addresses[voterIndex] == _voter);
+        address lastVoter = delegatedVoters[_delegate].addresses[delegatedVotersCount - 1];
+        if (voterIndex < delegatedVotersCount - 1) {
+            delegatedVoters[_delegate].addresses[voterIndex] = lastVoter;
+            delegates[lastVoter].voterIndex = voterIndex;
+        }
+        delegatedVoters[_delegate].addresses.length--;
+        emit ResetDelegate(_voter, _delegate);
```

Comment:
technically at this line delegate is not completely reset, since it still in the delegates mapping, I suggest to move events out of this func

Reply:
Maybe make ordering of voting and delegate the same?
` function _removeDelegatedAddressFor(address _delegate, address _voter)`
`emit ResetDelegate(_voter, _delegate)`